src/services/XmppSocket.js

```diff
@@ -283,6 +283,20 @@ export default {
       this.context.$store.setContactPresence({ jid: fullJid.bare, presence: presence.show, status: presence.status })
     })
 
+    // listen for retracted messages
+    this.client.on('messageRetracted', (retracted) => {
+      this.context.$store.updateMessage({
+        stanzaId: retracted.stanzaId,
+        // replace body and links
+        body: `Moderated by ${retracted.by.resource} (${retracted.reason})`,
```

Comment:
I assume there is always a `reason`

Reply:
I asked on the XSF channel and I haven't got any real answer for now, apart from someone agreeing this is probably optional. I am going to submit a PR to the spec to add explicit wording for this.

Reply:
I just checked [ConverseJS' code](https://github.com/conversejs/converse.js/blob/7539fc1a6f82233081bcbb0061e5694564775e48/src/headless/plugins/muc/muc.js#L830), JC being the author of the spec, and it seems to consider the reason optional.

Reply:
OK, I will add a control on this to avoid `Moderated by admin ()`

Reply:
Good call, in addition to the empty parentheses, following code (in `XmppClient.js`) was crashing when retract received without reason:
``` js
moderation[0].getChild('reason').getText()
```